templates/master-index.json

```diff
@@ -0,0 +1,58 @@
+[
+     {
+        "displayName": "OpenShift Node.js template",
+        "description": "Sample OpenShift Node.js application",
+        "language": "nodejs",
+        "projectType": "odo",
+        "projectStyle": "odo",
+        "location": "https://github.com/openshift/nodejs-ex"
+    },
+    {
+        "displayName": "OpenShift Python template",
+        "description": "Sample OpenShift Python application",
+        "language": "python",
+        "projectType": "odo",
+        "projectStyle": "odo",
+        "location": "https://github.com/OpenShiftDemos/os-sample-python"
+    },
+    {
+        "displayName": "OpenShift Ruby template",
+        "description": "Sample OpenShift Ruby application",
+        "language": "ruby",
+        "projectType": "odo",
+        "projectStyle": "odo",
+        "location": "https://github.com/OpenShiftDemos/os-sample-ruby"
+    },
+    {
+        "displayName": "OpenShift PHP template",
+        "description": "Sample OpenShift PHP application",
+        "language": "php",
+        "projectType": "odo",
+        "projectStyle": "odo",
+        "location": "https://github.com/OpenShiftDemos/os-sample-php"
+    },
+    {
+        "displayName": "OpenShift HTTPD template",
+        "description": "Sample OpenShift HTTPD application",
+        "language": "httpd",
+        "projectType": "odo",
+        "projectStyle": "odo",
+        "location": "https://github.com/OpenShiftDemos/http-base"
+    },
+    {
+        "displayName": "OpenShift Wildfly template",
+        "description": "Sample OpenShift Wildfly application",
+        "language": "wildfly",
+        "projectType": "odo",
+        "projectStyle": "odo",
+        "location": "https://github.com/OpenShiftDemos/openshift-tasks"
+    },
+    {
+        "displayName": "OpenShift Swift template",
+        "description": "Runtime for Swift applications",
+        "language": "swift",
+        "projectType": "odo",
+        "projectStyle": "odo",
+        "location": "https://"
+    }
+]
```

Comment:
```
[root@codewind-workspacelbxixnzo5l5ot71g-7d9948745c-dx5gs bin]# ./odo catalog list components
Odo Supported OpenShift Components:
NAME       PROJECT       TAGS
nodejs     openshift     10,8,8-RHOAR,latest

OpenShift Components:
NAME        PROJECT       TAGS
dotnet      openshift     2.0,latest
httpd       openshift     2.4,latest
nginx       openshift     1.10,1.12,1.8,latest
nodejs      openshift     6
perl        openshift     5.24,5.26,latest
php         openshift     7.0,7.1,latest
python      openshift     2.7,3.5,3.6,latest
ruby        openshift     2.3,2.4,2.5,latest
wildfly     openshift     10.0,10.1,11.0,12.0,13.0,8.1,9.0,latest
```
From the OpenShift components of the output above, I think we can remove `swift` and find & add `dotnet`, `nginx` and `perl` into the `master-index.json `

Reply:
Otherwise we may have the issue where user can choose component type from IDE but we don't provide the actual template.

Reply:
I suggest we add new line at the end of the file to avoid potential issues if we wan to process/parse the file in the future.

Reply:
I just added the swift template in there to test that our templatesProvider.js actually filters it out in the final index.json, since swift is not on the list returned by odo.